### ISSUE TITLE
manifestKeyMap error throwed after version upgrade

### ISSUE THREAD

**Issue by oleghalin** [brillout/vite-plugin-ssr#596]
### Description

After package version bump facing issue with Cloudflare workers deployment. 

```
Uncaught (in promise) TypeError: Cannot set property manifestKeyMap of #<Object> which has only a getter
```

### Error Message + Error Stack

```shell
Object.assign(obj, objAddendum);
         ^
      at objectAssign
  (/Users/u/Projects/frontend/node_modules/.pnpm/vite-plugin-ssr@0.4.69_vite@3.2.5/node_modules/vite-plugin-ssr/dist/cjs/utils/objectAssign.js:6:11)
      at assertPluginManifest
  (/Users/u/Projects/frontend/node_modules/.pnpm/vite-plugin-ssr@0.4.69_vite@3.2.5/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/manifest/assertPluginManifest.js:15:30)
      at initGlobalContext
  (/Users/u/Projects/frontend/node_modules/.pnpm/vite-plugin-ssr@0.4.69_vite@3.2.5/node_modules/vite-plugin-ssr/dist/cjs/node/globalContext.js:41:57)
```

Current version: `/vite-plugin-ssr/0.4.69_vite@3.2.5:`
Previous version: `  /vite-plugin-ssr/0.4.54_vite@3.2.4:`

**Comment by brillout** [comment]
Seems more like a bug on Cloudflare's side?

**Comment by oleghalin** [comment]
Not sure, version downgrading resolves this issue

**Comment by brillout** [comment]
I know, still, it seems like something funky on Cloudflare's side.

Actually, I've an idea why this can be. I'll have a look at it later today.

**Comment by brillout** [comment]
I can't reproduce with https://github.com/brillout/vite-plugin-ssr/tree/main/examples/cloudflare-workers-react. It seems to be working fine. What's your `$ wrangler --version`?

**Comment by brillout** [comment]
The one I'm using is `2.0.21`. If your app fails with that same version, then I'll need a reproduction (ideally a minimal one).

**Comment by brillout** [comment]
I'll reopen once I get a minimal reproduction.

**Comment by spacedawwwg** [comment]
I've got the same issue upgrading from 0.4.54 to ^0.4.55 using Netlify functions to server side render

**Comment by spacedawwwg** [comment]
It's blowing up when hitting `renderPage` in our netlify function.

reverted back to 0.4.54 while we investigate

```import type { Handler, HandlerEvent } from '@netlify/functions';
import { builder } from '@netlify/functions';
import { fixPathSlashes } from '@unilever/utils-helpers';
import consola from 'consola';
import { renderPage } from 'vite-plugin-ssr';

export async function generatePage(event: HandlerEvent) {
  const { path, headers } = event;
  const fixedPath = fixPathSlashes(path);

  // If there is an origin set the Access control
  const CORSDomain = headers?.origin || '*';

  consola.info('Request to path:', path);

  if (path !== fixedPath) {
    consola.info('Redirecting to fixed path:', fixedPath);
    return {
      statusCode: 301,
      headers: {
        Location: fixedPath,
      },
    };
  }

  try {
    consola.info('Setting page context path:', fixedPath);

    const pageContextInit = { urlOriginal: fixedPath };

    const pageContext = await renderPage(pageContextInit);

    const { httpResponse } = pageContext;
    
    if (!httpResponse) {
      throw new Error('No http response from renderPage pageContext');
    }
    
    const { body, statusCode, contentType } = httpResponse;

    return {
      statusCode,
      headers: {
        'Content-Type': contentType,
        'Access-Control-Allow-Origin': CORSDomain,
        'X-Robots-Tag': 'none',
        'Access-Control-Allow-Credentials': 'true',
      },
      body,
    };
  } catch (error) {
    return {
      statusCode: error.httpStatusCode || 500,
      body: JSON.stringify(
        {
          error: error.message,
        },
        null,
        2
      ),
    };
  }
}

export const handler = builder(generatePage as Handler);
```

**Comment by spacedawwwg** [comment]
Looking at this, I have netlify + vps working in a minimal repro.

What this is leading me to believe is that it is our custom pre-renderer that is blowing up now - so there may have been a change from 0.4.54 to 0.4.55 that needs to be addressed (looking at the release notes now)

**Comment by spacedawwwg** [comment]
@brillout narrowed it down to this one line:

https://github.com/brillout/vite-plugin-ssr/blob/22c9fb8bf3ba8c44c1049fb1b48cbe274563cf46/vite-plugin-ssr/node/plugin/plugins/manifest/assertPluginManifest.ts#L36

If I comment out the the objectAssign (as manifestKeyMap is a only a getter and has no setter) the render function works again.

Do you have any ideas here?

**Comment by brillout** [comment]
I think I've enough information now, I may be able to debug this without a reproduction.

**Comment by brillout** [comment]
> in our netlify function.

PR welcome to add a Netlify example to `examples/` so that every commit is tested against Netlify. Does Netlify have a preview thing like Cloudflare's miniflare preview?

**Comment by spacedawwwg** [comment]
> > in our netlify function.
> 
> PR welcome to add a Netlify example to `examples/` so that every commit is tested against Netlify. Does Netlify have a preview thing like Cloudflare's miniflare preview?

100% if you like, I never did it because https://vite-plugin-ssr.com/netlify-functions already had an example (albeit outdated) here

**Comment by spacedawwwg** [comment]
> I think I've enough information now, I may be able to debug this without a reproduction.

```TS
const { manifestKeyMap } = pluginManifest;
const pluginManifestClone = Object.assign({}, pluginManifest);
assert(isStringRecord(manifestKeyMap))
objectAssign(pluginManifestClone, { manifestKeyMap })
checkType<PluginManifest>(pluginManifestClone)
```

This fixed things

**Comment by brillout** [comment]
👍 It's only useful though if there is an official way to simulate/run a real-ish Netlify prod env. FYI that's why there isn't a `examples/vercel/` because I'm not aware of any way to preview the Vercel prod end.

**Comment by spacedawwwg** [comment]
I can set up an example that explains the user would need to install netlify CLI to test it locally, which is pretty straight forward to be honest

**Comment by brillout** [comment]
If VPS's CI can run it then it's good. (I'd assume it to be the case.)

**Comment by spacedawwwg** [comment]
> If VPS's CI can run it then it's good. (I'd assume it to be the case.)

You'd need an auth token is all: https://github.com/netlify/actions/blob/master/cli/README.md

**Comment by brillout** [comment]
Fix released in `0.4.70`.
